packages/components/src/internal/util/utils.ts

```diff
@@ -309,15 +309,23 @@ function isSameWithStringCompare(value1: any, value2: any): boolean {
  * @param updatedValues an object mapping fieldKeys to values that are being updated
  * @param primaryKeys the list of primary fieldKey names
  */
-export function getUpdatedData(originalData: Map<string, any>, updatedValues: any, primaryKeys: List<string>): any[] {
+export function getUpdatedData(
+    originalData: Map<string, any>,
+    updatedValues: any,
+    primaryKeys: List<string>,
+    additionalPkCols?: string[]
```

Comment:
nit: Consider describing this param in the function's documentation (since we've already gone through the process of describing these parameters).